_posts/en/newsletters/2020-12-09-newsletter.md

```diff
@@ -0,0 +1,230 @@
+  over the P2P network without using centralized directories.
+
+    Developers from both implementations discussed options, with most
+    participants seeming to prefer changing the other team's software.
+    The main issue seems to revolve around the negotiation of new
+    protocol features, which was previously discussed related to another
+    protocol change (see Newsletters [#111][news111 negotiation] and
+    [#118][news118 negotiation]).  The following is our rough summary of
+    what seem to be the dominant published opinions of individual
+    contributors to the different implementations:
+
+    - *Bitcoin Core* developers would like to upgraded nodes to announce
+      their support for new P2P protocol features by sending an empty
+      message named for the feature, e.g. `sendaddrv2` for the addrv2
+      feature and `wtxidrelay` for the [wtxid relay][] feature.  They would
+      like to stop using the P2P protocol version number to determine
+      what new features should be used with a particular peer.  They're
+      concerned that decentralized development of multiple
+      implementations cannot ensure that all nodes above a certain
+      protocol number will support all the features of lower numbers;
+      e.g. if Bitcoin Core were to increase it's protocol version number
+      to 80002, that doesn't mean it should be forced to implement the
+      xthin blocks protocol developed with version number 80001 for
+      Bitcoin Unlimited.
+
+    - *Libbitcoin* has a developer who still [wants][voskuil versioning]
+      version numbers to be incremented for new features even if an
+      alternative method is used for the actual negotiation.  He also
+      dislikes the empty named messages negotiation technique Bitcoin
+      Core has implemented; he instead favors of appending data to the
+      existing `verack` message.
+
+    - *Btcd* has a developer who also [wants][osuntokun versioning]
+      version numbers to be incremented so that btcd nodes don't receive
+      any unexpected message types, believing unexpected types to
+      present a denial of service attack vector.  Developers of other
+      software [claim][corallo risk] this isn't a major risk and that
+      it's easy to create problems of similar minor severity against any
+      network-connected system even if it's not running a Bitcoin node.
+
+    In the short term, a [PR][Bitcoin Core #20564] has been opened to
+    prevent Bitcoin Core from sending the `sendaddrv2` message to nodes
+    with older protocol versions.  In the long term, we hope a feature
+    negotiation protocol can be found that's acceptable to nearly all
+    developers.
+
+- **Thwarted upgrade to uppercase bech32 QR codes:** BTCPay Server
+  recently implemented an optimization described in [BIP173][] (covered
+  in [Newsletter #46][bech32 upper]) of creating QR codes for bech32
+  addresses that start with all capital letters in order to create
```

Comment:
"start with" ⇒ shouldn't the whole address be capitalized?

Reply:
@Xekyo See https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses for details including efficiency estimates, but the problem is that the query parameters in BIP21 URIs are case sensitive, so the best you can to do is something like `BITCOIN:BC1FOO?amount=1.00000000&label=more_money_more_problems`.  I'll turn this into a link in the text for anyone else with the same question, thanks for questioning!

Reply:
Oh, I see, I wasn't thinking of the query parameters. Thanks for clarifying. That write-up you linked to is stupendous.